homeassistant/components/geonetnz_quakes/geo_location.py

```diff
@@ -97,7 +92,7 @@ def __init__(
 
     async def async_added_to_hass(self) -> None:
         """Call when entity is added to hass."""
-        if self.hass.config.units.name == CONF_UNIT_SYSTEM_IMPERIAL:
+        if self.hass.config.units.length_unit == LENGTH_MILES:
```

Comment:
Checking the `name` of the unit system will stop work if we rename or add a new unit system.
It is more future-proof to check the `length_unit` of the unit system.

Reply:
Adjusted to use `hass.config.units == ***`